commands/device/list-devices.ts

```diff
@@ -1,18 +1,26 @@
-import { createTable } from "../../helpers";
+import { createTable, formatListOfNames } from "../../helpers";
 
 export class ListDevicesCommand implements ICommand {
 	constructor(private $devicesService: Mobile.IDevicesService,
+		private $errors: IErrors,
+		private $emulatorHelper: Mobile.IEmulatorHelper,
 		private $logger: ILogger,
 		private $stringParameter: ICommandParameter,
 		private $mobileHelper: Mobile.IMobileHelper,
-		private $emulatorImageService: Mobile.IEmulatorImageService,
 		private $options: ICommonOptions) { }
 
 	public allowedParameters = [this.$stringParameter];
 
 	public async execute(args: string[]): Promise<void> {
 		if (this.$options.availableDevices) {
-			await this.$emulatorImageService.listAvailableEmulators(this.$mobileHelper.validatePlatformName(args[0]));
+			const platform = this.$mobileHelper.normalizePlatformName(args[0]);
+			if (!platform && args[0]) {
```

Comment:
Because when `tns devices --available-devices` command was executed, an error was thrown from `validatePlatformName` method. I fixed this behaviour.
Actually when the platform is not valid (for example when `tns devices invalidplatform`), `normalizePlatformName` method will return `undefined` and the check above 
```
if (!platform && args[0]) {
```
will thrown an error.